civitai-downloader-v2/src/core/error/unified_error_handler.py

```
@@ -0,0 +1,472 @@
+            except Exception as fallback_error:
+                # Both services failed
+                self.error_statistics.record_error(
+                    category=self.categorize_error(fallback_error).value,
+                    error_type=type(fallback_error).__name__,
+                    component=service_name,
+                    operation=operation,
+                    message=f"Fallback service also failed: {str(fallback_error)}"
+                )
+                
+                # Re-raise the original error
+                raise e
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)